src/extensions/coblocks-settings/coblocks-settings-store.js

```diff
@@ -1,17 +1,75 @@
 /*global coblocksSettings*/
-
 /**
  * WordPress dependencies
  */
-import apiFetch from '@wordpress/api-fetch';
+import { createReduxStore, register, registerStore } from '@wordpress/data';
 import { applyFilters } from '@wordpress/hooks';
+import apiFetch from '@wordpress/api-fetch';
+
+/**
+ * Constant will check if `@wordpress/data` has deprecated `registerStore`
+ * by checking if `createReduxStore` or `register` are undefined
+ *
+ * @constant {boolean} registerStoreDeprecated
+ */
+const registerStoreDeprecated =
+	typeof createReduxStore !== 'undefined' &&
+	typeof register !== 'undefined';
+
+/**
+ * Helper function for the CoBlocks Settings Store.
+ * This function is used to detect which data-stores are deprecated
+ * and use the current supported data-store type.
+ *
+ * @function getStoreType
+ * @param {string} storeName - Store identifier for CoBlocks Setting store.
+ * @param {Object} storeOptions - Object with Redux-like configuration.
+ * @return {Function} createReduxStore(storeName, storeOptions) || registerStore(storeName, storeOptions)
+ */
+const getStoreType = ( storeName, storeOptions ) => {
+	return registerStoreDeprecated
+		? createReduxStore( storeName, storeOptions ) : registerStore( storeName, storeOptions );
+};
```

Comment:
This comment applies to the entire file. Honestly nothing needs to be changed here for the 5.7 release since the functions are deprecated not removed. Instead we should look at refactoring this store in the future while also bumping the minimum WP version to 5.7 (because of the new APIs).

I think the best thing we could do here is remove this store entirely and instead use the core data api for interacting with the settings endpoint. This only requires we register our settings with the REST API on the PHP side.

Reply:
I'd like a bit more information here on what you are suggesting. Are you referring to making API PUT requests to handle settings changes and exclude the use of a store altogether? I would agree with that solution if so.

One of the reasons that I had to change the store completely was that WordPress 5.7 had not just deprecated the APIs but had also made enough sweeping changes to break backward compatibility for the deprecated APIs. The store, as we were using it, was no longer functional.

Reply:
Here is the behavior of CoBlocks settings store prior to refactors: https://dashboard.cypress.io/projects/sovnn2/runs/1172/test-results/d3772cfb-af9f-4d09-b0bf-c088fdf8d6b7

Essentially UI would not reflect actually saved settings.

Reply:
Logically the old function still exists and works with no changes from my tests and that test failure seems to be caused by the selector not existing. I'm not seeing that selector actually being rendered although it does exist in our code. Are we using the component properly?

Regardless, we can still merge this I'm just noting this is a deprecation of the core API not a breaking change and that we should remove this store in the future.